### DISCUSSION TITLE
How to perform type casting in prisma?

### DISCUSSION THREAD

**Discussion by Josrph** [prisma/prisma#17248] [Q&A] [answered]
is it possible to cast a column from one type to another using prisma?

**for example:** cast an id column of type integer to a string then perform a contains query on the new string column.

**Comment by nurul3101** [comment, marked as answer]
Hello @Josrph 👋 

Prisma queries will return data in the JavaScript types that [correspond](https://www.prisma.io/docs/concepts/database-connectors/postgresql#type-mapping-between-postgresql-to-prisma-schema) to the Prisma types defined in your `schema.prisma`. It’s possible to do SQL type casting, like casting from integer to string if you use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

**Reply by Josrph** [reply]
so the only option is to use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access).

**Reply by nurul3101** [reply]
At the moment, yes.

**Reply by enesbala5** [reply]
However you don't support $queryRaw in where statements - so this is not useful to most. Please let me know if there is something I'm not aware of. Thank you

This feature is being requested at #5560

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
